prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -341,16 +420,38 @@ def retrieve_top_datasets(
     def retrieve_dataset_dict(
         self,
         prompt_spec: PromptSpec,
+        data_transform: bool = False,
+        num_transform: int = 3000,
     ) -> datasets.DatasetDict | None:
         """Select a dataset from a prompt using a dual-encoder retriever.
 
         Args:
             prompt_spec: A prompt whose instruction field we use to retrieve datasets.
+            data_transform: Whether to transform the dataset or not.
```

Comment:
This variable description could be much more descriptive. What does it mean to transform a dataset? Why would you want to do this?